docker_config_generator/templates/golang

```diff
@@ -0,0 +1,26 @@
+FROM golang:1.20-alpine AS builder
+
+RUN apk update && apk --no-cache upgrade
+RUN apk --no-cache add ca-certificates git
+WORKDIR /build
+# Fetch dependencies
+COPY go.mod go.sum ./
+RUN go mod download
+
+COPY . .
+RUN CGO_ENABLED=0 ${BUILD_COMMAND} .
+
+FROM alpine:latest AS runner
+
+RUN apk --no-cache upgrade
+RUN mkdir /user  \
+    && adduser -D user --shell /usr/sbin/nologin \
+    && chown -R user:user /user
+WORKDIR /user
+
+COPY --from=builder /build/app .
```

Comment:
The output filename always will not be `app` always. Depends on the module name. I think we need to provide name in build command `-o name` and another build args for name to know the file name while copying.